.script/solutionValidator.ts

```
@@ -0,0 +1,128 @@
+            console.warn(`No "resources" field found in the file. Skipping file path: ${filePath}`);
+            return ExitCode.SUCCESS;
+        }
+
+        // get the resources from the file
+        let resources = jsonFile.resources;
+
+        // filter resources that have type "Microsoft.OperationalInsights/workspaces/providers/metadata"
+        const filteredResource = resources.filter(function (resource: { type: string; }) {
+            return resource.type === "Microsoft.OperationalInsights/workspaces/providers/metadata";
+        });
+        if (filteredResource.length > 0) {
+            filteredResource.forEach((element: { hasOwnProperty: (arg0: string) => boolean; properties: { hasOwnProperty: (arg0: string) => boolean; categories: any; }; }) => {
+                // check if the resource has a "properties" field
+                if (element.hasOwnProperty("properties") === true) {
+                    // check if the "properties" field has a "categories" field
+                    if (element.properties.hasOwnProperty("categories") === true) {
+                        const categories = element.properties.categories;
+
+                        let invalidDomains = [];
+                        let invalidVerticals = [];
+
+                        // check if the categories have a "domains" field
+                        if (categories.hasOwnProperty("domains")) {
+                            let domains = categories.domains;
+                            for (const domain of domains) {
+                                // check if the domain is valid
+                                if (!validDomains.includes(domain)) {
+                                    invalidDomains.push(domain);
+                                }
+                            }
+                        }
+
+                        // check if the categories have a "verticals" field
+                        if (categories.hasOwnProperty("verticals")) {
```

We need to fail in case either "verticals" property itself is not available or don't have at least 1 value ?

---

Got a confirmation that verticals can be optional.